modules/programs/zsh.nix

```diff
@@ -444,6 +436,17 @@ in
 
     {
       home.file."${relToDotDir ".zshenv"}".text = ''
+        ${cfg.envExtraFirst}
+
+        ${optionalString (cfg.dotDir != null) ''
+          ZDOTDIR=${zdotdir}
+        ''}
+
+        ${optionalString cfg.oh-my-zsh.enable ''
+          ZSH="${pkgs.oh-my-zsh}/share/oh-my-zsh";
+          ZSH_CACHE_DIR="${config.xdg.cacheHome}/oh-my-zsh";
+        ''}
```

Comment:
I'm not an OMZ user, so I haven't verified this order personally. But considering the variables in question, it seemed to me like the kind of early-init configuration that belongs towards the top of the file.